pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -186,6 +189,7 @@ def __init__(
         every_n_train_steps: Optional[int] = None,
         every_n_val_epochs: Optional[int] = None,
         period: Optional[int] = None,
+        trigger_on_train_end: bool = False,
```

Comment:
Shouldn't it default to True so the current behaviour is kept?

Reply:
@carmocca yes, it could be. we'd also need to make `save_last=True` by default too then right? otherwise we'd run into the warning commented about below